apps/openmw/engine.cpp

```diff
@@ -477,8 +477,20 @@ void OMW::Engine::prepareEngine (Settings::Manager & settings)
     }
 
     // find correct path to the game controller bindings
-    const std::string localdefault = mCfgMgr.getLocalPath().string() + "/gamecontrollerdb.txt";
-    const std::string globaldefault = mCfgMgr.getGlobalPath().string() + "/gamecontrollerdb.txt";
+    // File format for controller mappings is different for SDL <= 2.0.4, 2.0.5, and >= 2.0.6
+    SDL_version linkedSdlVersion;
+    SDL_GetVersion(&linkedSdlVersion);
+    std::string controllerFileName;
+    if (linkedSdlVersion.major == 2 && linkedSdlVersion.minor == 0 && linkedSdlVersion.patch <= 4) {
+        controllerFileName = "gamecontrollerdb_204.txt";
```

Comment:
If you already do such checks, maybe to make more generic solution: get the current SDL version, make a name of the appropriate file and to try to load it and if it is absent, use default file name? It will allow to avoid the same problems in the future.

Reply:
SDL 2.0.6 and 2.0.7 have the same format for controller mappings. Hopefully that means we are out of the woods for controller formats and can drop the old ones in the future. But even if 2.0.8 comes out and changes it again, then we'll have to do some logic to have one file for both 2.0.6 and 2.0.7.

Also, it's been a long time since I worked with files in C++ and all I remember with that was that it was a real pain. Knowing that the format is the same for two consecutive versions, do you still recommend going that route? If so, could you give me a pointer on how to go about doing it?